ts/db_test.go

```diff
@@ -0,0 +1,351 @@
+					} else {
+						tm.t.Logf("\tActual value: %v", its)
+					}
+					if its, err := proto.InternalTimeSeriesDataFromValue(vModel); err != nil {
+						tm.t.Logf("\tModel value is not a valid time series: %v", vModel)
+					} else {
+						tm.t.Logf("\tModel value: %v", its)
+					}
+				}
+			}
+		}
+
+		// Detect keys in model which were not present in the actual data.
+		for k := range tm.modelData {
+			n, s, r, ts := DecodeDataKey([]byte(k))
+			if _, ok := actualData[k]; !ok {
+				tm.t.Logf("Key %s/%s@%d, r:%d from model was not found in actual data", n, s, ts, r)
+			}
+		}
+
+		tm.t.Fatalf("Failing because model data was not equal to actual data.")
+	}
+}
+
+// AssertKeyCount asserts that the model contains the expected number of keys.
+// This is used to ensure that data is actually being generated in the test
+// model.
+func (tm *tsTestModel) AssertKeyCount(expected int) {
+	if a, e := len(tm.modelData), expected; a != e {
+		tm.t.Errorf("model data key count did not match expected value: %d != %d", a, e)
+	}
+}
+
+// StoreTimeSeriesData instructs the model to store the given time series data
+// in both the model and the system under test.
+func (tm *tsTestModel) StoreTimeSeriesData(r Resolution, data proto.TimeSeriesData) {
+	// Store data in the system under test.
+	if err := tm.db.storeData(r, data); err != nil {
+		tm.t.Fatalf("error storing time series data: %s", err.Error())
+	}
+
+	// Process and store data in the model.
+	internalData, err := data.ToInternal(r.KeyDuration(), r.SampleDuration())
+	if err != nil {
+		tm.t.Fatalf("test could not convert time series to internal format: %s", err.Error())
+	}
+
+	for _, idata := range internalData {
+		key := MakeDataKey(data.Name, data.Source, r, idata.StartTimestampNanos)
+		keyStr := string(key)
+
+		existing, ok := tm.modelData[keyStr]
+		var newTs *proto.InternalTimeSeriesData
+		if ok {
+			existingTs, err := proto.InternalTimeSeriesDataFromValue(existing)
+			if err != nil {
+				tm.t.Fatalf("test could not extract time series from existing model value: %s", err.Error())
+			}
+			newTs, err = engine.MergeInternalTimeSeriesData(existingTs, idata)
+			if err != nil {
+				tm.t.Fatalf("test could not merge time series into model value: %s", err.Error())
+			}
+		} else {
+			newTs, err = engine.MergeInternalTimeSeriesData(idata)
+			if err != nil {
+				tm.t.Fatalf("test could not merge time series into model value: %s", err.Error())
+			}
+		}
+		val, err := newTs.ToValue()
+		if err != nil {
+			tm.t.Fatal(err)
+		}
+		tm.modelData[keyStr] = val
+	}
+}
+
+// TestStoreTimeSeries is a simple test of the Time Series module, ensuring that
+// it is storing time series correctly.
+func TestStoreTimeSeries(t *testing.T) {
+	tm := newTsTestModel(t)
+	tm.Start()
+	defer tm.Stop()
+
+	// Basic storage operation: one data point.
+	tm.StoreTimeSeriesData(Resolution10s, proto.TimeSeriesData{
+		Name: "test.metric",
+		Datapoints: []*proto.TimeSeriesDatapoint{
```

Comment:
Your tests would become slightly more concise if you added `intDatapoint` and `floatDatapoint` helper methods which took a timestamp and value and returned a `*proto.TimeSeriesDatapoint`.